File: Editor/src/main/java/org/turbomedia/turboedit/editor/windows/NewProjectWindow.java
package org.turbomedia.turboedit.editor.windows;

import org.turbomedia.turboedit.editor.panes.NewProjectPane;
import javafx.scene.Scene;
import javafx.stage.Modality;
import javafx.stage.Stage;

import static org.turbomedia.turboedit.editor.Editor.ICON;
import static org.turbomedia.turboedit.editor.Editor.TITLE;
import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public class NewProjectWindow extends Stage {

    public NewProjectWindow(Stage stage) {
        getIcons().add(ICON);
        setTitle(TITLE + " - " + GetText("title.new_project"));
        setAlwaysOnTop(true);
        setResizable(false);

        int HEIGHT = 250;
        setMaxHeight(HEIGHT);
        setHeight(HEIGHT);

        int WIDTH = 450;
        setMaxWidth(WIDTH);
        setWidth(WIDTH);

        setScene(new Scene(new NewProjectPane(this, stage)));

        initModality(Modality.APPLICATION_MODAL);
        showAndWait();
    }
}

File: Editor/src/main/kotlin/turboedit/editor/windows/NewProjectWindow.kt
package turboedit.editor.windows

import javafx.scene.Scene
import javafx.stage.Modality
import javafx.stage.Stage
import turboedit.editor.Editor
import turboedit.editor.misc.Locale
import turboedit.editor.panes.NewProjectPane

class NewProjectWindow(stage: Stage?) : Stage() {
    init {
        icons.add(Editor.Companion.ICON)
        title = Editor.Companion.TITLE + " - " + Locale.getText("title.new_project")
        isResizable = false
        isAlwaysOnTop = true

        val height = 250
        maxHeight = height.toDouble()
        setHeight(height.toDouble())

        val width = 450
        maxWidth = width.toDouble()
        setWidth(width.toDouble())

        scene = Scene(NewProjectPane(this, stage))

        centerOnScreen()
        initModality(Modality.APPLICATION_MODAL)
        showAndWait()
    }
}
